scripts/mem_usage.py

```diff
@@ -0,0 +1,149 @@
+import re
+import subprocess
+import sys
+
+def get_args():
+
+    parser = argparse.ArgumentParser(
+                description='Shows the memory usage of a OP-TEE based '
+			'on ELF sections')
+    parser.add_argument('tee_elf', help='the OP-TEE ELF file (tee.elf)')
+    parser.add_argument('-a', '--all', action='store_true', help=' same as '
+                        '-i -p -u -U')
+    parser.add_argument('-n', '--no-map', action='store_true', help=' do not '
+                        'show the detailed section mappings and RAM usage')
+    parser.add_argument('-i', '--init', action='store_true', help='report '
+                        'the total size of the .*_init sections')
+    parser.add_argument('-p', '--paged', action='store_true', help='report '
+                        'the total size of the .*_pageable sections')
+    parser.add_argument('-u', '--unpaged', action='store_true',
+                        help='report the total size of the unpaged sections, '
+                        'that is, all sections but the ones in --init or '
+                        '--paged')
+    parser.add_argument('-U', '--unpaged-no-heap', action='store_true',
+                        help='report the size of all unpaged sections '
+                        'excluding heap space. Reflects the size of unpaged '
+                        'code and data (.text, .rodata, .data, .bss, .nozi '
+                        'and possibly unwind tables)')
+    parser.add_argument('-r', '--raw', action='store_true', help='when '
+                        'processing -i, -p, -u, or -U, show only the size (in '
+                        'decimal) and no other text')
+
+    return parser.parse_args()
+
+def printf(format, *args):
+    sys.stdout.write(format % args)
+
+def print_sect(name, addr, size, round_up, print_num_pages):
+    if args.no_map:
+        return
+    if size == 0:
+        size_kib = 0;
+        num_pages = 0;
+    else:
+        if round_up:
+            size_kib = (size - 1) / 1024 + 1
+        else:
+            size_kib = size / 1024
+        num_pages = (size - 1) / 4096 + 1
+
+    printf('%-16s %.8X - %.8X size %.8X %3d KiB', name, addr, addr + size,
+           size, size_kib)
+    if print_num_pages:
+        printf(' %d pages', num_pages)
+    printf('\n')
+
+def print_pager_stat(name, size):
+    size_kib = size / 1024
+    if args.raw:
+        printf('%d ', size)
+    else:
+        printf('%-36s size %.8X %3d KiB\n', name, size, size_kib)
+
+def main():
+
+    global args
+
+    in_shdr = False
+    sects = []
+    init_size = 0
+    paged_size = 0
+    unpaged_size = 0
+    unpaged_no_heap_size = 0
+
+    args = get_args()
+
+    readelf = subprocess.Popen(['readelf', '-a', '-W', args.tee_elf],
+                               stdout = subprocess.PIPE)
+    for line in iter(readelf.stdout.readline, ''):
+        if 'Section Headers:' in line:
+            in_shdr = True
+            continue
+        if 'Key to Flags:' in line:
```

Comment:
These labels mandates the English LANG. Maybe should insure LANG=en_GB.UTF-8 or equivalent when invoking the readelf tool.